### PR TITLE
Site/auto deploy

### PR DESCRIPTION
**Pull request by clairesunstudio**
Added:
  - project: Site
    component: CircleCI
    description: Add CI for site for build and deploy. (#1001)
    issue: DP-17980
    impact: Patch

---

Added documentation for branch name prefix
| Branch name prefix  | Rule | Scenario |
|---|---|---|
| `react/`  | This will bypass Patternlab build, deploy and tests | This prefix should only be used if all your changes are in the [react](/react) folder, and nothing in the shared [assets](/assets) folder that could potentially impact the build of the Patternlab implementation  |
| `pattern/`  | This will bypass React Storybook build and tests | This prefix should only be used if all your changes are in the [patternlab](/patternlab) folder, and nothing in the shared [assets](/assets) folder that could potentially impact the build of the React implementation  |
| `site/`  | This will bypass all React and Patternlab and assets build, deploy and tests. |  Only use this prefix if all the changes are in the [site](/site) folder. e.g. Content changes to the homepage site that won't impact React, Patternlab and shared assets. |
| `docs`  | This will bypass all React and Patternlab and assets build, deploy and tests, as well as Mayflower site build and tests. | Only use this prefix if all the changes are in the [docs](/docs) folder. e.g. Documentation changes for the repo and the Mayflower Gitbook that won't impact React, Patternlab, shared assets and the homepage static site. |

### CODE REVIEW

**Comment by clairesunstudio** [comment]
Not sure why Github is still showing `patternlab_test` and `react_test` being required. filters are added to the jobs in 58b9478

File: .circleci/config.yml [hunk 1]
```diff
@@ -395,3 +452,8 @@ workflows:
           filters:
             branches: { ignore: /.*/ }
             tags: { only: /.*/ }
+      - site_deploy_tag:
+          requires: [site_build]
```
Comment (l0c0luke) [inline .circleci/config.yml:460]: Need to add `site_build` job to the tag push workflow.
```
#!/bin/sh -eo pipefail
# site_deploy_tag: requires job \"site_build\" but \"site_build\" is not part of this workflow.
# The following jobs are unreachable: site_deploy_tag
```
from https://app.circleci.com/pipelines/github/massgov/mayflower/52/workflows/fabda5d4-b9d7-461e-88ae-bf63a1f0d17a/jobs/25838

**Review by l0c0luke — CHANGES_REQUESTED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (clairesunstudio) [reply .circleci/config.yml:460]: good catch!

**Comment by l0c0luke** [comment]
@clairesunstudio `patternlab_test` and `react_test` are mandatory jobs as defined in this repo @rbayliss showed me. It's not a bad idea to run these jobs all the time anyways, just to be sure, so you should remove the ignoring of them.

**Comment by clairesunstudio** [comment]
@lukewertz I removed the restrictions from github settings to allow bypassing the tests for none code changes, and added documentation in README.md to specify the use the branch prefixes. Hopefully it makes more sense now.

**Review by l0c0luke — APPROVED** [review]

**Pull request merged**